src/components/search.js

```diff
@@ -0,0 +1,44 @@
+import { useState, useEffect, useRef } from 'react';
+import axios from 'axios';
+
+const Search = (props) => {
+    const [searchValue, setSearchValue] = useState("");
+    const inputRef = useRef();
+    const { onSelectingMovie } = props;
+
+    
+  useEffect(() => {
+    const timer = setTimeout(() => {
+      if(searchValue === inputRef.current.value){
+        axios.get(`https://api.themoviedb.org/3/search/movie?api_key=ed4f70d18fa8cbc03fb59ce04847e1bc&query=${searchValue}`)
```

Comment:
same API key is being used in multiple components, store it in a different file as constant and import where required